broker/tasks/alb.py

```diff
@@ -91,3 +96,34 @@ def remove_certificate_from_alb(operation_id, **kwargs):
         )
     db.session.add(service_instance)
     db.session.commit()
+
+
+@huey.retriable_task
+def remove_certificate_from_previous_alb(operation_id, **kwargs):
+    operation = Operation.query.get(operation_id)
+    service_instance = operation.service_instance
+    remove_certificate = Certificate.query.filter(
+        and_(
+            Certificate.service_instance_id == service_instance.id,
+            Certificate.id != service_instance.current_certificate_id,
+        )
+    ).first()
+
+    operation.step_description = "Removing SSL certificate from load balancer"
+    db.session.add(operation)
+    db.session.commit()
+
+    time.sleep(int(config.DNS_PROPAGATION_SLEEP_TIME))
+
+    if service_instance.previous_alb_listener_arn is not None:
+        alb.remove_listener_certificates(
+            ListenerArn=service_instance.previous_alb_listener_arn,
+            Certificates=[
+                {"CertificateArn": remove_certificate.iam_server_certificate_arn}
+            ],
+        )
```

Comment:
I'm just making sure I understand the flow correctly here - this is removing the previous certificate from the ALB, right?

Reply:
correct